### PR TITLE
Add Aws launch configuration resource

### PR DESCRIPTION
**Pull request by dmurray-chef**
### Description
Add Aws Launch Configuration resource, terraform, control, tests and docs

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_launch_configuration.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+title: About the aws_launch_configuration Resource
+---
+
+# aws\_launch\_configuration
+
+Use the `aws_launch_configuration` InSpec audit resource to test properties of a single AWS Launch Configuration. 
+
+<br>
+
+## Availability
```
Comment (r-fennell) [inline docs/resources/aws_launch_configuration.md]: You can delete the Availability, Installation and Version headers

File: docs/resources/aws_launch_configuration.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+### Installation
+
+### Version
+
+
+## Syntax
+
+    # Ensure that a queue exists and has a visibility timeout of 300 seconds
```
Comment (r-fennell) [inline docs/resources/aws_launch_configuration.md]: Is this the correct description?

File: docs/resources/aws_launch_configuration.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    describe aws_launch_configuration('my-config') do
+      it { should exist }
```
Comment (r-fennell) [inline docs/resources/aws_launch_configuration.md]: Tab out to align :)

File: libraries/aws_launch_configuration.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      @instance_type               = configuration[:instance_type]
+      @iam_instance_profile        = configuration[:iam_instance_profile]
+      @key_name                    = configuration[:key_name]
+      @security_groups             = configuration[:security_groups]
+      @associate_public_ip_address = configuration[:associate_public_ip_address].nil? ? false : true
+      @ebs_optimized               = configuration[:ebs_optimized]
+      @spot_price                  = configuration[:spot_price].to_f
+      @instance_monitoring         = configuration[:instance_monitoring][:enabled] ? 'detailed': 'basic'
+      @user_data                   = configuration[:user_data] ? Base64.decode64(configuration[:user_data]) : nil
+    end
```
Comment (r-fennell) [inline libraries/aws_launch_configuration.rb]: I like the use of ternary here.

File: docs/resources/aws_launch_configuration.md (hunk 1 above)
Comment (dmurray-chef) [reply docs/resources/aws_launch_configuration.md]: 👍 done

File: docs/resources/aws_launch_configuration.md (hunk 2 above)
Comment (dmurray-chef) [reply docs/resources/aws_launch_configuration.md]: 👍 fixed

File: libraries/aws_launch_configuration.rb (hunk 4 above)
Comment (dmurray-chef) [reply libraries/aws_launch_configuration.rb]: 👍 Fixed

File: docs/resources/aws_launch_configuration.md (hunk 3 above)
Comment (dmurray-chef) [reply docs/resources/aws_launch_configuration.md]: 👍 Fixed

**Pull request merged**